pkg/controller/incarnation/incarnation_controller.go

```diff
@@ -438,21 +307,12 @@ func (r *ReconcileIncarnation) Reconcile(request reconcile.Request) (reconcile.R
 		health = false
 	}
 
-	incarnation.Status = picchuv1alpha1.IncarnationStatus{
-		Health: picchuv1alpha1.IncarnationHealthStatus{
-			// TODO(bob): when we get metrics, evaluate actual health
-			Healthy: health,
-			Metrics: []picchuv1alpha1.IncarnationHealthMetricStatus{},
-		},
-		Scale: picchuv1alpha1.IncarnationScaleStatus{
-			Current: replicaset.Status.ReadyReplicas,
-			Desired: replicaset.Status.Replicas,
-		},
-		Resources: resourceStatus,
-	}
+	incarnation.Status.Health.Healthy = health
+	incarnation.Status.Scale.Current = replicaset.Status.ReadyReplicas
+	incarnation.Status.Scale.Desired = replicaset.Status.Replicas
+	incarnation.Status.Resources = resourceStatus
 	if e := r.client.Status().Update(context.TODO(), incarnation); e != nil {
 		reqLogger.Error(e, "Failed to update Incarnation status")
 	}
-
 	return reconcile.Result{}, err
```

Comment:
this can return `Requeue: true` with a delay and then picchu will keep the status up to date